### ISSUE TITLE
'is not an array' on update (upsert)

### ISSUE THREAD

**Issue by jiku** [aldeed/meteor-collection2#255]
While

```
Coll.upsert(selector, { $set: { data: [data] }})
```

with

```
CollSchema = new SimpleSchema
  data:
    type: [Object]
    label: 'Data'
    optional: true
    blackbox: true

Coll.attachSchema CollSchema
```

Works great on insert, adding data: [data] to the document where data is an Array with one element, 0, it fails on update, with this

```
Error: Data must be an array
```

I have tried

```
dataArray = []
dataArray.push data

Coll.upsert(selector, { $set: { data: dataArray }})
```

and made sure data is an array in all sorts of ways (_.isArray, instanceof, etc.)
Updates also work when not attaching the schema.
I don't want to manipulate [data] at all. Just replace it.

**Comment by aldeed** [comment]
Try using the expanded definition. I'm not sure `blackbox` is understood correctly when you use the `[Object]` shortcut.

```coffeescript
CollSchema = new SimpleSchema
  data:
    type: Array
    label: 'Data'
    optional: true
  'data.$':
    type: Object,
    blackbox: true
```

**Comment by jiku** [comment]
Thanks for the quick response! Fixed the issue and think it was just an oversight on my part. Another Coll.upsert being done in a callback right after that used an $elemMatch selector on data to update another top-level field. Just added ```validate: false``` to that.

With blackbox, I tried expanding the definition first and that made no difference. The shortcut worked on the first insert, and now works with upserts too. With data now having several elements and their content being very plastic, so it seems to work.
